ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -59,7 +65,9 @@ TensorLayout::TensorLayout(DataType dtype, const PageConfig& page_config, const
 }
 
 TensorLayout TensorLayout::fromLegacyPaddedShape(DataType dtype, const PageConfig& page_config, const MemoryConfig& memory_config, const ttnn::Shape& legacy_shape) {
-    return TensorLayout(dtype, page_config, memory_config, CMAKE_UNIQUE_NAMESPACE::legacyShapeToAlignment(legacy_shape));
+    auto result =  TensorLayout(dtype, page_config, memory_config);
+    result.alignment_ = CMAKE_UNIQUE_NAMESPACE::legacyShapeToAlignment(legacy_shape);
+    return result;
```

Comment:
This will get default alignment and validate on it (I didn't really think through this logic if `legacyShapeToAlignment` is expected to be called later). Before it would just validate the passed in alignment from `legacyShapeToAlignment`. Is there a reason for this change?

Reply:
I think there was, looking into it more now

Reply:
The reason for this change is that otherwise `initialize_alignment` -> `create_default_alignment` would be called. Currently it creates a lot of issues across the codebase. I think this would disappear when we migrate all of our code to not use `fromLegacyPaddedShape`

Reply:
`initialize_alignment` -> `create_default_alignment` would happen after this change not before right?